src/transactions/utils.ts

```diff
@@ -551,7 +551,15 @@ export const convertFromSa = {
         amountAsset: input.data.amount.asset.id,
         priceAsset: input.data.price.asset.id,
       },
-      price: input.data.price.getCoins(),
+      price: input.data.price
+        .getTokens()
+        .mul(
+          new BigNumber(10).pow(
+            8 +
+              input.data.price.asset.precision -
+              input.data.amount.asset.precision
+          )
+        ),
```

Comment:
Based on the specification, the difference in decimals is required for version 1 and 2 
https://docs.waves.tech/en/blockchain/transaction-type/exchange-transaction#json-representation

Probably should also fix these checks in the tests

Reply:
Oh, looked at the wrong section of the documentation. 
It means everything is right.
https://docs.waves.tech/en/blockchain/order#json-representation